integration/poetry_with_unavailable_version_test.go

```diff
@@ -0,0 +1,73 @@
+package integration_test
```

Comment:
In general we try to avoid integration tests for failure cases. There's a large number of ways the buildpacks can fail, and driving them out with integration tests is slow.

If we really wanted to ensure that the version of python selected was provided by `pyproject.toml` then we can expand the integration test for the successful case (see comments below) to more rigorously validate that.